src/blocks/Block.as

```diff
@@ -134,6 +134,8 @@ public class Block extends Sprite {
 		} else if (type == "b") {
 			base = new BlockShape(BlockShape.BooleanShape, color);
 			isReporter = true;
+			forceAsync = Scratch.app.extensionManager.shouldForceAsync(op);
+			isRequester = forceAsync;
```

Comment:
Compare to the handling for `'r'` blocks below, which are the same as `'b'` other than the return type and the graphical shape of the block.